### PR TITLE
Update URLs in comments, messages, and docs

### PR DESCRIPTION
**Pull request by gguillotte**
Several changes to the docs site have broken URLs used in comments and documentation. While many of these URLs redirect to appropriate locations, others do not, and current redirects might not always work as expected. Additionally, some links point to docs that no longer exist or are relevant.

### CODE REVIEW

**Comment by jhelwig** [comment]
It's safe to ignore the failing Travis check, since it's in the `warnings` rake task, and the things it's complaining about are pre-existing issues in one of the acceptance tests that got touched.

```
acceptance/tests/parser_functions/puppet_lookup_cmd.rb
acceptance/tests/parser_functions/puppet_lookup_cmd.rb:2181: warning: assigned but unused variable - rep
acceptance/tests/parser_functions/puppet_lookup_cmd.rb:2211: warning: assigned but unused variable - re4
acceptance/tests/parser_functions/puppet_lookup_cmd.rb:2241: warning: assigned but unused variable - repm4
acceptance/tests/parser_functions/puppet_lookup_cmd.rb:2271: warning: assigned but unused variable - re1m4
acceptance/tests/parser_functions/puppet_lookup_cmd.rb:2301: warning: assigned but unused variable - re2m4
acceptance/tests/parser_functions/puppet_lookup_cmd.rb:2362: warning: assigned but unused variable - re4m4
```

File: lib/puppet/module_tool/skeleton/templates/generator/examples/init.pp.erb [hunk 1]
```diff
@@ -7,6 +7,6 @@
 # environment (to compare the resulting system state to the desired state).
 #
 # Learn more about module testing here:
-# https://docs.puppet.com/guides/tests_smoke.html
+# https://puppet.com/docs/puppet/latest/modules_fundamentals.html#writing-modules
```
Comment (jhelwig) [inline lib/puppet/module_tool/skeleton/templates/generator/examples/init.pp.erb]: It wasn't immediately obvious where to go to read more about module testing when I followed the new link to the Modules Fundamentals doc. Would https://puppet.com/docs/puppet/5.5/bgtm.html#testing-your-module be a better link to use here? (I got to the Beginning Guide to Modules section from the links in the Modules Fundamentals doc, but only after clicking around hoping to find something that directly mentioned module testing.)

File: lib/puppet/parser/functions/hiera_include.rb [hunk 2]
```diff
@@ -89,7 +89,7 @@ module Puppet::Parser::Functions
 result must be post processed to get exactly the same result, for example using simple hash/array `+` or
 with calls to stdlib's `deep_merge` function depending on kind of hiera call and setting of merge in hiera.yaml.
 
-See [the documentation](http://links.puppet.com/hierainclude) for more information
+See [the documentation](https://puppet.com/docs/puppet/latest/hiera_intro.html#assigning-classes-to-nodes-with-hiera-hierainclude) for more information
```
Comment (jhelwig) [inline lib/puppet/parser/functions/hiera_include.rb]: This anchor doesn't seem to exist on the page?

File: lib/puppet/parser/functions/with.rb [hunk 3]
```diff
@@ -3,9 +3,9 @@
   :type => :rvalue,
   :arity => -1,
   :doc => <<-DOC
-Call a [lambda](https://docs.puppetlabs.com/puppet/latest/reference/lang_lambdas.html)
+Call a [lambda](https://puppet.com/docs/puppet/latest/lang_lambdas.html)
 with the given arguments and return the result. Since a lambda's scope is
-[local](https://docs.puppetlabs.com/puppet/latest/reference/lang_lambdas.html#lambda-scope)
+[local](https://puppet.com/docs/puppet/latest/lang_lambdas.html#lambda-scope)
```
Comment (jhelwig) [inline lib/puppet/parser/functions/with.rb]: Anchor doesn't appear to be on the page?

File: lib/puppet/module_tool/skeleton/templates/generator/examples/init.pp.erb (hunk 1 above)
Comment (gguillotte) [reply lib/puppet/module_tool/skeleton/templates/generator/examples/init.pp.erb]: Good call! I'll see if we can get that original URL redirected to the BGTM as well.

File: lib/puppet/parser/functions/hiera_include.rb (hunk 2 above)
Comment (gguillotte) [reply lib/puppet/parser/functions/hiera_include.rb]: We probably shouldn't link to deprecated features' docs anyway, so I'll replace with a link to the Hiera-to-lookup migration guide.

File: lib/puppet/parser/functions/with.rb (hunk 3 above)
Comment (gguillotte) [reply lib/puppet/parser/functions/with.rb]: :+1:

**Comment by gguillotte** [comment]
@jhelwig Addressed feedback.

**Pull request merged**